### PR TITLE
[DOCS] Expands DFA and TM API docs with required privileges info

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR edits the required security privileges in the data frame analytics and the trained models API docs. It also removes links that point to sec privileges, built-in roles, and ML security setup pages from the prereq section.

Related to https://github.com/elastic/ml-team/issues/525

**Applied principles:**
* document the least privilege
* always mention cluster privilege
* mention the built-in role that grants the respective cluster privilege
* mention index privileges when it is necessary
* use flowing-text if there is only one privilege
* use bullet points if there are more privileges

### Preview

[Delete DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/delete-dfanalytics.html)
[Delete TM aliases](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/delete-trained-models-aliases.html)
[Delete TM](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/delete-trained-models.html)
[Evaluate DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/evaluate-dfanalytics.html)
[Explain DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/explain-dfanalytics.html)
[GET DFA stats](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-dfanalytics-stats.html)
[GET DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-dfanalytics.html)
[GET TM starts](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-trained-models-stats.html)
[GET TM](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/get-trained-models.html)
[Preview DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/preview-dfanalytics.html)
[PUT DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-dfanalytics.html)
[PUT TM aliases](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-trained-models-aliases.html)
[PUT TM](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/put-trained-models.html)
[Start DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/start-dfanalytics.html)
[Stop DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/stop-dfanalytics.html)
[Update DFA](https://elasticsearch_71335.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/update-dfanalytics.html)

### CODE REVIEW

File: docs/reference/ml/df-analytics/apis/delete-dfanalytics.asciidoc [hunk 1]
```diff
@@ -23,6 +23,7 @@ If the {es} {security-features} are enabled, you must have the following
 built-in roles or equivalent privileges:
 
 * `machine_learning_admin`
+* cluster: `manage_ml` or `manage`
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/delete-dfanalytics.asciidoc:26]: Should the `machine_learning_admin` entry above this line be in the same line as `cluster`?

File: docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc [hunk 2]
```diff
@@ -23,7 +23,8 @@ Evaluates the {dfanalytics} for an annotated index.
 If the {es} {security-features} are enabled, you must have the following 
 privileges:
 
-* cluster: `monitor_ml`
+* cluster: `monitor_ml` or `monitor`
+* index: `read`, `write`
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/evaluate-dfanalytics.asciidoc:27]: this should only be `read`

File: docs/reference/ml/df-analytics/apis/stop-dfanalytics.asciidoc [hunk 3]
```diff
@@ -27,6 +27,7 @@ If the {es} {security-features} are enabled, you must have the following
 built-in roles or equivalent privileges:
 
 * `machine_learning_admin`
+* cluster: `manage_ml` or `manage`
```
Comment (dimitris-athanasiou) [inline docs/reference/ml/df-analytics/apis/stop-dfanalytics.asciidoc:30]: Same here with `machine_learning_admin` above?

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

File: docs/reference/ml/df-analytics/apis/delete-dfanalytics.asciidoc [hunk 4]
```diff
@@ -19,12 +19,8 @@ Deletes an existing {dfanalytics-job}.
 [[ml-delete-dfanalytics-prereq]]
 == {api-prereq-title}
 
-If the {es} {security-features} are enabled, you must have the following 
-built-in roles or equivalent privileges:
-
-* `machine_learning_admin`
-
-For more information, see <<built-in-roles>> and {ml-docs-setup-privileges}.
+It is required to have the `manage_ml` cluster privilege to use this API. This
```
Comment (lcawl) [inline docs/reference/ml/df-analytics/apis/delete-dfanalytics.asciidoc:22]: If you want to align with the style of the introductory sentence, I think this also works:
```suggestion
Requires the `manage_ml` cluster privilege. This 
```

**Review by lcawl — APPROVED** [review]
LGTM!

**Pull request merged**